Packages/Tracking/Physical Hands/Runtime/Scripts/UI/PhysicalHandsButtonToggle.cs

```diff
@@ -0,0 +1,175 @@
+/******************************************************************************
+ * Copyright (C) Ultraleap, Inc. 2011-2024.                                   *
+ *                                                                            *
+ * Use subject to the terms of the Apache License 2.0 available at            *
+ * http://www.apache.org/licenses/LICENSE-2.0, or another agreement           *
+ * between Ultraleap and you, your company or other organization.             *
+ ******************************************************************************/
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace Leap.Unity.PhysicalHands
+{
+    /// <summary>
+    /// Class for toggling a button using physical hands.
+    /// </summary>
+    public class PhysicalHandsButtonToggle : PhysicalHandsButtonBase
+    {
+        private bool _canUnpress = false;
+        private RigidbodyConstraints _constaintsBeforeFreeze = RigidbodyConstraints.None;
+
+        public bool IsToggled
+        {
+            get
+            {
+                return _isButtonPressed;
+            }
+            set
+            {
+                _isButtonPressed = value;
+            }
```

Comment:
Should this also handle the functions below to actual set the toggle state?

Reply:
possibly, id maybe just get rid of the set function all together